### PR TITLE
[Feat]: find API with missing parameters

### PR DESCRIPTION
**Pull request by dakabang**
In our inference service, we found that the find API will have better performance by directly writing the missed keys when the hit rate is high, which it usually is.

### CODE REVIEW

**Comment by dakabang** [comment]
In our inference service, we found that the find API will have better performance by directly writing the missed keys when the hit rate is high, which it usually is.

File: include/merlin/core_kernels/lookup.cuh [hunk 1]
```diff
@@ -21,6 +21,44 @@
 namespace nv {
 namespace merlin {
 
+template <typename K>
+struct FoundFunctorV1 {
+  __host__ __device__ FoundFunctorV1(bool* __restrict founds_)
+      : founds(founds_) {}
+
+  __forceinline__ __device__ void operator()(const int idx, const K /*key*/,
+                                         const bool found) {
```
Comment (rhdong) [inline include/merlin/core_kernels/lookup.cuh:30]: Please format the code by referring the [guidance](https://github.com/NVIDIA-Merlin/HierarchicalKV/blob/master/STYLE_GUIDE.md)

Comment (dakabang) [reply include/merlin/core_kernels/lookup.cuh]: fixed

File: include/merlin/core_kernels/lookup.cuh [hunk 2]
```diff
@@ -1017,24 +1163,53 @@ __global__ void lookup_kernel(const Table<K, V, S>* __restrict table,
...
+}
+
+template <class K, class V, class S, uint32_t TILE_SIZE = 4>
+__global__ void lookup_kernel(
+    const Table<K, V, S>* __restrict table, Bucket<K, V, S>* buckets,
+    const size_t bucket_max_size, const size_t buckets_num, const size_t dim,
+    const K* __restrict keys, V** __restrict values, S* __restrict scores,
+    K* __restrict missed_keys, int* __restrict missed_indices,
+    int* __restrict missed_size, int* __restrict dst_offset, size_t N) {
+  FoundFunctorV2<K> found_functor(missed_keys, missed_indices, missed_size);
```
Comment (MoFHeka) [inline include/merlin/core_kernels/lookup.cuh:1208]: atomicadd in FoundFunctorV2 runs on all warps when the key is not found, which can cause performance degradation when there are lots of missed key. Isn't it better to split the missed_keys size used by found_functor from n to the bucket size, and then merge according to missed_size at last kernel finishing?

Comment (dakabang) [reply include/merlin/core_kernels/lookup.cuh:1208]: Good point, I'll work on refining this code.

Comment (MoFHeka) [reply include/merlin/core_kernels/lookup.cuh:1208]: > Good point, I'll work on refining this code.

What was the result? Does this optimization show up in performance tests?

Comment (dakabang) [reply include/merlin/core_kernels/lookup.cuh:1208]: I apologize for the late reply, but I have verified your suggestion. Due to the additional calculations brought by the bucket separation, the benchmark results did not show performance improvement.

**Pull request merged**